### DISCUSSION TITLE
Why not making semicolons optional

### DISCUSSION THREAD

**Discussion by NaveedAhmadHematmal** [dotnet/csharplang#5655] [Language Ideas] [answered]
In the last versions of c#, the team made c# syntax easy. I can name a few like file-scoped namespaces, global usings, and...

Removing semicolons or making them optional can help and it will be more English-like. It will compete with python.

**Comment by HaloFour** [comment, marked as answer]
#1581
#496

https://mobile.twitter.com/jaredpar/status/1010195445887336450

**Reply by TahirAhmadov** [reply]
It's almost like there needs to be a pinned discussion or something which says "Yes optional semicolons were proposed a million times already and no, it's not happening".

**Reply by jnm2** [reply]
This is literally the example that is brought up each time the idea of a discussion intro document comes up.

**Comment by huoyaoyuan** [comment]
> Why not making semicolons optional

Because **new line is optional** in C family languages. You can remove almost every new line, the only exceptions are `#` directives and `//` comments.

> can help and it will be more English-like

Why and how? In English article you write a period for every sentence.

> It will compete with python

Why? The quality of a language isn't decided by such trivial things.

**Comment by Xyncgas** [comment]
無駄無駄無駄無駄無駄
